### PR TITLE
ENG-18667: use the EDS executor to parallelize SSM operations

### PR DESCRIPTION
**Pull request by rdykiel**
Each SSM may use a different executor service

### CODE REVIEW

**Comment by manjujames** [comment]
The changes look good to me. However, I am not sure how much better this will be because we are still talking about a lot of zk back and forth.

File: src/frontend/org/voltcore/zk/SynchronizedStatesManager.java [hunk 1]
```diff
@@ -88,7 +88,8 @@
     private Set<String> m_groupMembers = new HashSet<String>();
     private final StateMachineInstance m_registeredStateMachines [];
     private int m_registeredStateMachineInstances = 0;
-    private static final ListeningExecutorService m_shared_es = CoreUtils.getListeningExecutorService("SSM Daemon", 1);
+    private static final ListeningExecutorService s_shared_es = CoreUtils.getListeningExecutorService("SSM Daemon", 1);
+    private final ListeningExecutorService m_ssm_es;
```
Comment (n3world) [inline src/frontend/org/voltcore/zk/SynchronizedStatesManager.java]: Minor: This should be camel case other than the `m_` prefix same with other variable. The name es alone doesn't tell you much why not call it something like `executor` it is only a little longer and more descriptive than just `es`?
Also, why put the initials of this class in a member variable name isn't that redundant.

File: src/frontend/org/voltcore/zk/SynchronizedStatesManager.java [hunk 2]
```diff
@@ -1891,6 +1902,7 @@ public SynchronizedStatesManager(ZooKeeper zk, String rootPath, String ssmNodeNa
 
     // Used only for Mocking StateMachineInstance
     public SynchronizedStatesManager() {
```
Comment (n3world) [inline src/frontend/org/voltcore/zk/SynchronizedStatesManager.java]: Not your change but having public constructors for test mocking seems a bit suspicious.

File: src/frontend/org/voltcore/zk/SynchronizedStatesManager.java [hunk 3]
```diff
@@ -1925,8 +1949,12 @@ public SynchronizedStatesManager(ZooKeeper zk, String rootPath, String ssmNodeNa
         m_memberId = m_canonical_memberId + "_v" + m_resetCounter;
     }
 
+    public ListeningExecutorService getExecutor() {
```
Comment (wweiss-voltdb) [inline src/frontend/org/voltcore/zk/SynchronizedStatesManager.java]: Why is this method here?

File: src/frontend/org/voltcore/zk/SynchronizedStatesManager.java (hunk 2 above)
Comment (rdykiel) [reply src/frontend/org/voltcore/zk/SynchronizedStatesManager.java]: Will not change

File: src/frontend/org/voltcore/zk/SynchronizedStatesManager.java (hunk 1 above)
Comment (rdykiel) [reply src/frontend/org/voltcore/zk/SynchronizedStatesManager.java]: Renamed variables

File: src/frontend/org/voltcore/zk/SynchronizedStatesManager.java (hunk 3 above)
Comment (rdykiel) [reply src/frontend/org/voltcore/zk/SynchronizedStatesManager.java]: Thought I would need it; removed

**Comment by rdykiel** [comment]
Closing due to risk of blocking site threads; superseded by changes to make SSM initialization asynchronous.

**Pull request closed without merging**